// react-text-transition/lib/components/TextTransition.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var web_1 = require("@react-spring/web");
function TextTransition(props) {
    var _a = props.direction, direction = _a === void 0 ? 'up' : _a, _b = props.inline, inline = _b === void 0 ? false : _b, _c = props.springConfig, springConfig = _c === void 0 ? web_1.config.default : _c, _d = props.delay, delay = _d === void 0 ? 0 : _d, className = props.className, style = props.style, _e = props.translateValue, tv = _e === void 0 ? '100%' : _e, children = props.children;
    var initialRun = (0, react_1.useRef)(true);
    var fromTransform = direction === 'down' ? "-".concat(tv) : tv;
    var leaveTransform = direction === 'down' ? tv : "-".concat(tv);
    var transitions = (0, web_1.useTransition)([children], {
        enter: { opacity: 1, transform: 'translateY(0%)' },
        from: { opacity: 0, transform: "translateY(".concat(fromTransform, ")") },
        leave: {
            opacity: 0,
            transform: "translateY(".concat(leaveTransform, ")"),
            position: 'absolute',
        },
        config: springConfig,
        immediate: initialRun.current,
        delay: !initialRun.current ? delay : undefined,
    });
    var _f = (0, react_1.useState)(0), width = _f[0], setWidth = _f[1];
    var currentRef = (0, react_1.useRef)(null);
    var heightRef = (0, react_1.useRef)('auto');
    (0, react_1.useEffect)(function () {
        initialRun.current = false;
        var element = currentRef.current;
        if (!element)
            return;
        var _a = element.getBoundingClientRect(), width = _a.width, height = _a.height;
        setWidth(width);
        heightRef.current = height;
    }, [children, setWidth, currentRef]);
    var widthTransition = (0, web_1.useSpring)({
        to: { width: width },
        config: springConfig,
        immediate: initialRun.current,
        delay: !initialRun.current ? delay : undefined,
    });
    return (react_1.default.createElement(web_1.animated.div, { className: "text-transition ".concat(className), style: __assign(__assign(__assign({}, (inline && !initialRun.current ? widthTransition : undefined)), style), { whiteSpace: inline ? 'nowrap' : 'normal', display: inline ? 'inline-flex' : 'flex', height: heightRef.current }) }, transitions(function (styles, item) { return (react_1.default.createElement(web_1.animated.div, { style: __assign({}, styles), ref: item === children ? currentRef : undefined, children: item })); })));
}
exports.default = TextTransition;
